test/dist_autograd_test.py

```diff
@@ -13,6 +13,11 @@
 prev_rank_rpc_done = False
 prev_rank_context_id = 0
 
+known_context_ids = []
+
+def store_context_id(context_id):
+    global known_context_ids
+    known_context_ids.append(context_id)
```

Comment:
Add a small comment here mentioning we don't need a lock here since we hold the GIL when executing python UDFs.